dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiRequestFilter.java

```diff
@@ -0,0 +1,41 @@
+package io.rtr.jsonapi.filter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.rtr.jsonapi.JsonAPIRequest;
+import org.glassfish.jersey.message.internal.MediaTypes;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.Provider;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+@Provider
+public class JsonApiRequestFilter implements ContainerRequestFilter
+{
+
+  public static final MediaType JSONAPI_MEDIATYPE = MediaType.valueOf("application/vnd.api+json");
+  @Override
+  public void filter(ContainerRequestContext requestContext)
+    throws IOException
+  {
+    if (requestContext.hasEntity() && MediaTypes.typeEqual(JSONAPI_MEDIATYPE, requestContext.getMediaType()))
+    {
+      InputStream in = requestContext.getEntityStream();
+      if (in.getClass() != ByteArrayInputStream.class) {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
This isn't the same object mapper configured by Dropwizard. It may deserialize things differently than the service serializes them. I think there is a way to get the ObjectMapper, but I don't know it off the top of my head. Will look into it.